### ISSUE TITLE
cucumber-expressions: regexp bug - parentheses in character class regex

### ISSUE THREAD

**Issue by spicalous** [cucumber/cucumber#454]
## Summary

A regular expression for a step definition is not generating the correct step parameters
 
## Current Behavior

I have the following regexp for a step in my project
~~~
/^an example step: ([A-Z_, ()]+)$/
~~~
Should match and return correct parameters for
`an example step: VALUE_ONE, VALUE_TWO (ABC)`

Currently it matches, but returns undefined for the step parameter

## Possible Solution

We are able to change our regexp to match step in passing test (mentioned below), however I feel like this should still be fixed as it used to work in older version of cucumber

## Steps to Reproduce (for bugs) 

I have managed to produce a failing test in 
https://github.com/cucumber/cucumber-expressions-javascript

Add the following lines to https://github.com/cucumber/cucumber-expressions-javascript/blob/master/test/regular_expression_test.js
~~~
  it.only('this test fails', function() {
    const parameterRegistry = new ParameterTypeRegistry()
    const expr = /^the chart has settings with following drawings: ([A-Z_, ()]+)$/
    const expression = new RegularExpression(expr, parameterRegistry)

    const args = expression.match(
      'the chart has settings with following drawings: TREND_LINE, TREND_LINE (MACD)'
    )

    assert.equal(args[0].getValue(), 'TREND_LINE, TREND_LINE (MACD)')
  })

  it.only('this test passes', function() {
    const parameterRegistry = new ParameterTypeRegistry()
    const expr = /^the chart has settings with following drawings: (.*)$/
    const expression = new RegularExpression(expr, parameterRegistry)

    const args = expression.match(
      'the chart has settings with following drawings: TREND_LINE, TREND_LINE (MACD)'
    )

    assert.equal(args[0].getValue(), 'TREND_LINE, TREND_LINE (MACD)')
  })
~~~

## Context & Motivation

We're upgrading from a really old version of cucumber-js (0.5.3)
This used to work with the old version and seems like a regression

## Your Environment

Cucumber@4.2.1 in the browser

Any help would be appreciated
Thank you in advance!

**Comment by aslakhellesoy** [comment]
Thanks for the very detailed bug report @spicalous! I have an idea what the problem might be. I'll be working on a fix.

**Comment by spicalous** [comment]
Thanks @aslakhellesoy ! Let me know if I can be of any help :)

**Comment by spicalous** [comment]
@aslakhellesoy Could you help me confirm if there is also an issue in the underlying `becke-ch--regex--s0-0-v1--base--pl--lib` Regex library?

It seems like when I create a simple capturing group
~~~
const Regex = require('becke-ch--regex--s0-0-v1--base--pl--lib')
console.log(new Regex(/c (cucumber)/).exec('c (cucumber)'))
console.log(new Regex(/c (.*)/).exec('c (cucumber)'))
~~~
both produce different output

In chrome, this would have the same output
~~~
console.log('c cucumber'.match(/c (cucumber)/));
console.log('c cucumber'.match(/c (.*)/));
~~~

**Comment by aslakhellesoy** [comment]
@spicalous See above - I've reproduced and fixed the bug (in Java). Do you think you can port it to some of the other languages?

**Comment by spicalous** [comment]
@aslakhellesoy Nice! I will give it a go tomorrow evening, thanks! 👍

**Comment by spicalous** [comment]
#463 JS implemention added! :)

**Comment by luke-hill** [comment]
Wasn't this fixed in 6.0.2?

**Comment by aslakhellesoy** [comment]
You're right @luke-hill - my bad
